libs/core/langchain_core/callbacks/manager.py

```diff
@@ -314,12 +315,13 @@ def handle_event(
                 # If we try to submit this coroutine to the running loop
                 # we end up in a deadlock, as we'd have gotten here from a
                 # running coroutine, which we cannot interrupt to run this one.
-                # The solution is to create a new loop in a new thread.
-                with ThreadPoolExecutor(1) as executor:
-                    executor.submit(
-                        cast("Callable", copy_context().run), _run_coros, coros
-                    ).result()
+                # The solution is to run the synchronous function on the globally shared
+                # thread pool executor to avoid blocking the main event loop.
+                _executor().submit(
+                    cast("Callable", copy_context().run), _run_coros, coros
+                ).result()
```

Comment:
I think we still need to wait for the result() here but we could speed things up a lot if we didn't

Reply:
@hinthornw is there any parallelism a higher level (i.e., at the handle_event level) via either sync or async paths? (i.e., does the threadpool need more than 1 executor?)

Reply:
That's the right quesiton to ask and I'm going to have to double check the code